File: backend/game/modelslib/gamepiece.py
from django.db import models

# Create your models here.

class GamePiece(models.Model):
    game = models.ForeignKey(
        'Game',
        related_name = 'pieces',
        on_delete = models.CASCADE,
    )
    orientation = models.ForeignKey(
        'GamePieceOrientation',
        on_delete = models.CASCADE,
    )
    collectable_item = models.ForeignKey(
        'CollectableItem',
        on_delete = models.CASCADE,
        null=True
    )
    order = models.IntegerField(null=True)

File: frontend/app/src/reducers/gamePiece.js
const gamePiece = (state = {
    id: 0,
    game: 0,
    orientation: null,
    collectable_item: null,
    order: null
}, action) => {
  switch(action.type) {
    case "GAME_PIECE_UPDATE":
      const { id, game, orientation, collectable_item, order } = action.payload;
      return {
        ...state,
        id,
        game,
        orientation,
        collectable_item,
        order
      }
    default:
      return state;
  }
}

export default gamePiece;
